add Unit support in Spectrum conditions, add manual Hitran cross-sections support

erwanp opened the pull request:
## New features

- [x] Parse dimensioned values (i.e. with units) in Spectrum conditions
- [x] Parse and generate Spectrum from manually downloaded HITRAN cross section files (https://hitran.org/xsc/)

Done :

- [x] Add tests
- [x] Add Gallery example

Note : this PR might raise errors when units are not defined in user-scripts; in particular `path_length` and `pressure`. ```Spectrum.cond_units``` should be used to define them. 


## Example 

```python

from radis import Spectrum
from radis.phys.units import Unit as u

# Here we use a downloaded cross-section file from the test suite :
from radis.test.utils import getTestFile

datafile = getTestFile("CH3COCH3_233.4_375.2_700.0-1780.0_13.xsc")

# Temperature and pressure are given by the cross-section file used (above: 233.4 K; 375.2 Torr)
# Mole fraction and path length can be recomputed arbitrarily :
conditions = {"mole_fraction": 0.1, "path_length": 20 * u("mm")}

s = Spectrum.from_xsc(datafile, conditions)
s.name = "original L = 20mm"

s.plot("transmittance_noslit", lw=2)

# Rescale for other path lengths, and plot :
s.rescale_path_length(100 * u("mm"))
s.name = "rescaled L=100 mm"
s.plot(nfig="same")

import matplotlib.pyplot as plt

plt.title(s.c["molecule"])
plt.legend()
```


![image](https://github.com/user-attachments/assets/8205a568-f4a1-4deb-b4cf-567f8cade038)



---

Note @HajimeKawahara the cross-section parsing is added directly in the API  (hitranapi)

erwanp commented:
@EkulRF have a look and confirm this is what you need; then we'll merge

EkulRF commented:
Thanks @erwanp. This is what I need! Getting the error: 'Test file `CH3COCH3_297.8_700.0_700.0-1780.0_13.xsc` does not exist'

The .xsc files are stored in a local directory and correctly pathed. What should I modify for this to be solved?

EkulRF commented:
Thanks @erwanp 

I get the following error when I have the relevant xsc files saved to a local directory:

`datafile = getTestFile(filepath)`

`Exception has occurred: FileNotFoundError
Test file `~/CH3COCH3_297.8_700.0_700.0-1780.0_13.xsc` does not exist. Choose one of: 
- hitran_co2_626_bandhead_4165_4200nm.par
- geisa_CO2_fragment.par
- co2_cdsd_hamiltonian_fragment.levels
- geisa_CO_fragment.par
- N2C_specair_380nm.spec
- synth-NH3-1-500-2000cm-P10-mf0.01-p1.spec
- cdsd_hitemp_09_header.txt
- CO2_measured_spectrum_4-5um.spec
- geisa_H2O_fragment.par
- hitran_co_3iso_2000_2300cm.par
- hitran_CO_fragment.par
- CO_Tgas1500K_mole_fraction0.01.spec
- Corentin_0_100cm_DownSampled_20cm_10pctCO2_1-wc-gw450-gr300-sl1500-acc5000-.spec
- CO_Tgas1500K_mole_fraction0.5.spec
- CO2abscoeff_300K_4150_4400nm.txt
- geisa_O2_fragment.par
- hitran_2016_H2O_2iso_2000_2100cm.par
- cdsd_hitemp_09_fragment.txt
- calc_N2C_spectrum_Trot1200_Tvib3000.txt
- hitran_CO2_fragment.par
- slitfunction.txt
- calc_N2C_spectrum_Trot1200_Tvib3000_slit0.1.txt or use force=True`


The file is correctly pathed, etc. Am I missing something or have I stored the .xsc file incorrectly?

minouHub commented:
@erwanp any update on this one?

minouHub commented:
@EkulRF, it's been a long time :) 
You need to download manually the cross-section. Here is a more explicit version of the example proposed by @erwanp. This feature will be merged in 0.15.1 (in 1 week) or in 0.16 (planned for end of Spetember).

Here is an updated version: 
```python
from radis import Spectrum
from radis.phys.units import Unit as u

### In this example:
# The cross-section "CH3COCH3_233.4_375.2_700.0-1780.0_13.xsc" was manually downloaded by RADIS developper
# For another cross-section: 1/perform manual download from HITRAN and 2/ adapt `path_to_datafile`
from radis.test.utils import getTestFile

path_to_datafile = getTestFile("CH3COCH3_233.4_375.2_700.0-1780.0_13.xsc")
# Temperature and pressure are given by the cross-section file used above: 233.4 K; 375.2 Torr
# Mole fraction and path length can be recomputed arbitrarily :
conditions = {"mole_fraction": 0.1, "path_length": 20 * u("mm")}

s = Spectrum.from_xsc(path_to_datafile, conditions)

s.plot("transmittance_noslit", lw=2)

# Rescale for other path lengths, and plot :
s.rescale_path_length(100 * u("mm")).plot(nfig="same")

import matplotlib.pyplot as plt

plt.title(s.c["molecule"])
```

minouHub commented:
Still not merged. Please give me a sign if this feature is really needed.

erwanp commented:
The feature is good (combine our spectra with Hitran cross-sections) so let's implement. 

- I rebased on the main , update the PR text and the Example. 

- I got an error because of changes done in #683 to fix #681; I added a new way to fix the error in 68c8514080a1ed0fb37ee2e81a344725b8624911 which keeps this PR working

minouHub commented:
The tests not passing are solved in the develop branch. Taking a little risk by merging.
